agent/testflinger_agent/agent.py

```diff
@@ -111,6 +124,50 @@ def mark_device_offline(self):
         # Create the offline file, this should work even if it exists
         open(self.get_offline_files()[0], "w").close()
 
+    def unpack_attachments(self, job_data: dict, cwd: Path):
+        """Download and unpack the attachments associated with a job"""
+
+        # download attachment archive to a unique temporary folder
+        archive_dir = tmp_dir()
+        archive_path = self.client.get_attachments(
+            job_id=job_data["job_id"], path=archive_dir
+        )
+        if archive_path is None:
+            raise FileNotFoundError
+        # extract archive data to a unique temporary folder and clean up
+        extracted_dir = tmp_dir()
+        with tarfile.open(archive_path, "r:gz") as tar:
+            tar.extractall(extracted_dir, filter=secure_filter)
+        shutil.rmtree(archive_dir)
```

Comment:
I think we can do more to ensure this gets cleaned up. If we raise that FileNotFoundError above, or if anything else raises an exception, then the archive_dir will just linger around. I would suggest either using tempfile.TemporaryDirectory() as a context manager and converting the string path it gives you to a Path object if you need.
Otherwise, it might be possible to use a try/finally here to make sure we clean it up, but I generally like the context manager approach better since you don't have to know every possible exception